ledgerwallet/manifest_json.py

```diff
@@ -0,0 +1,78 @@
+import json
+import os
+from typing import Dict, Optional
+
+from ledgerwallet import params
+from ledgerwallet.manifest import AppManifest, icon_from_file
+
+
+class AppManifestJson(AppManifest):
+    def __init__(self, filename):
+        with open(filename) as f:
+            self.path = os.path.dirname(filename)
+            self.dic = json.load(f)
+            assert "targetId" in self.dic and "binary" in self.dic
+
+    @property
+    def app_name(self) -> str:
+        return self.dic["name"]
+
+    def data_size(self, device) -> int:
+        if "dataSize" not in self.dic:
+            return 0
+        else:
+            return self.dic["dataSize"]
+
+    def get_application_flags(self, device) -> int:
+        if "flags" not in self.dic:
+            return 0
+        else:
+            return int(self.dic["flags"], 16)
+
+    def get_api_level(self, device) -> int:
+        return int(self.dic["apiLevel"], 10)
```

Comment:
Given there is no check here, I assume `"apiLevel"` is a required field, so it should be checked in the `__init__` like `"targetId"` and `"binary"`

Reply:
Thanks again 

The `apiLevel` is not mandatory
There is the function `has_api_level` which allow to test the api level. 
Currently from what I see the only time this function is called the 'has_api_level' one is test just before: 
https://github.com/LedgerHQ/ledgerctl/blob/892bc543d8d3346011e9522605212292db401518/ledgerwallet/client.py#L319

Reply:
I see your next comment I will add some cheks